internal/pkg/deploy/cloudformation/stack/transformers.go

```diff
@@ -382,15 +382,17 @@ func convertELBAccessLogsConfig(mft *manifest.Environment) (*template.ELBAccessL
 }
 
 func convertFlowLogsConfig(mft *manifest.Environment) (*template.VPCFlowLogs, error) {
-	vpcFlowLogsArgs := mft.EnvironmentConfig.Network.VPC.FlowLogs.AdvancedFlowLogConfig
-	if !mft.VPCFlowLogsEnabled() {
+	vpcFlowLogs := mft.EnvironmentConfig.Network.VPC.FlowLogs
+	if !vpcFlowLogs.Basic && vpcFlowLogs.Advanced.Retention == nil {
```

Comment:
@dannyrandall I wodner if it's worthwhile updating the `Union` type's `IsZero` method? so that it returns true if the zero-value of the type is also true:
```go
	if t.IsBasic() {
		var zero Basic // NEW
		return isZeroerAndZero(t.Basic) ||  t.Basic == zero /* the conditional is new */
	}
	if t.IsAdvanced() {
		var zero Advanced
		return isZeroerAndZero(t.Advanced) || t.Advanced == zero
	}
	return true
```

this way we can replace this if statement with:
```go
if !vpcFlowLogs.IsZero() {
```

Reply:
sadly this doesn't work right now - because the type constraint is `any`. i think if we changed it to `comparable` it would work - and i think that's probably fine since we don't want interface types anyways 🤔 

i'll make this change in a separate PR @KollaAdithya!

Reply:
Sorry I just marked this as unresolved, I'm willing to give it a :shipit: with a do-not-merge label but I would love to see the `comparable` PR merged first so that this code is simplified before merging